src/jdk.javadoc/share/classes/jdk/javadoc/internal/doclets/formats/html/PackageWriterImpl.java

```diff
@@ -126,12 +141,64 @@ public Content getContentHeader() {
         return new ContentBuilder();
     }
 
+    private void computePackageData() {
+        relatedPackages = findRelatedPackages();
+        boolean isSpecified = utils.isSpecified(packageElement);
+        allClasses = filterClasses(isSpecified
+                ? utils.getAllClasses(packageElement)
+                : configuration.typeElementCatalog.allClasses(packageElement));
+    }
+
+    private SortedSet<TypeElement> filterClasses(SortedSet<TypeElement> types) {
+        List<TypeElement> typeList = types
+                .stream()
+                .filter(te -> utils.isCoreClass(te) && configuration.isGeneratedDoc(te))
+                .collect(Collectors.toList());
+        return utils.filterOutPrivateClasses(typeList, options.javafx());
+    }
+
+    private List<PackageElement> findRelatedPackages() {
+        String pkgName = packageElement.getQualifiedName().toString();
+
+        // always add super package
+        int lastdot = pkgName.lastIndexOf('.');
+        String pkgPrefix = lastdot > 0 ? pkgName.substring(0, lastdot) : null;
+        List<PackageElement> packages = new ArrayList<>(
+                filterPackages(p -> p.getQualifiedName().toString().equals(pkgPrefix)));
+
+        // add subpackages unless there are very many of them
+        Pattern subPattern = Pattern.compile(pkgName.replace(".", "\\.") + "\\.\\w+");
+        List<PackageElement> subpackages = filterPackages(
+                p -> subPattern.matcher(p.getQualifiedName().toString()).matches());
+        if (subpackages.size() <= MAX_SUBPACKAGES) {
+            packages.addAll(subpackages);
+        }
+
+        // only add sibling packages if we are beneath threshold, and number of siblings is beneath threshold as well
+        if (pkgPrefix != null && packages.size() <= MAX_SIBLING_PACKAGES) {
+            Pattern siblingPattern = Pattern.compile(pkgPrefix.replace(".", "\\.") + "\\.\\w+");
+
+            List<PackageElement> siblings = filterPackages(
+                    p -> siblingPattern.matcher(p.getQualifiedName().toString()).matches());
+            if (siblings.size() <= MAX_SIBLING_PACKAGES) {
+                packages.addAll(siblings);
+            }
+        }
```

Comment:
Hmmm, I foresee downstream JBS issues ... "javadoc bug: I added a package to my code and the table disappeared".  Should there be a warning and a way to change the limit? (Eventually?)